airflow/cli/commands/dag_command.py

```diff
@@ -294,38 +294,46 @@ def dag_next_execution(args):
 
     if dag.get_is_paused():
         print("[INFO] Please be reminded this DAG is PAUSED now.", file=sys.stderr)
+        with create_session() as session:
+            last_parsed_dag: DagModel = (
+                session.query(DagModel).filter(DagModel.dag_id == dag.dag_id)
+            ).one_or_none()
+        print(last_parsed_dag.next_dagrun.isoformat())
+    else:
+        with create_session() as session:
+            max_date_subq = (
+                session.query(func.max(DagRun.execution_date).label("max_date"))
+                .filter(DagRun.dag_id == dag.dag_id)
+                .subquery()
+            )
+            max_date_run: DagRun | None = (
+                session.query(DagRun)
+                .filter(DagRun.dag_id == dag.dag_id, DagRun.execution_date == max_date_subq.c.max_date)
+                .one_or_none()
+            )
 
```

Comment:
From what I can tell, lines below here should all be outside of the `with create_session()` block.

Reply:
I believe the below if block `if max_date_run is None:` does still belong within the `with create_session()` block based on previous commit. The code below that if block would be outside.